teamcity: failed tests on master: test/TestReplicaCommandQueueCancellation

The following tests appear to have failed:

[#42878](https://teamcity.cockroachdb.com/viewLog.html?buildId=42878):

```
--- FAIL: test/TestReplicaCommandQueueCancellation (0.019s)
Test ended in panic.
I161107 14:14:00.578517 2247 gossip/gossip.go:237  [n?] initial resolvers: []
W161107 14:14:00.578536 2247 gossip/gossip.go:1055  [n?] no resolvers found; use --join to specify a connected node
I161107 14:14:00.578569 2247 base/node_id.go:62  NodeID set to 1
I161107 14:14:00.578589 2247 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161107 14:14:00.579996 2247 storage/store.go:1174  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161107 14:14:00.580691 2251 storage/replica_proposal.go:328  [s1,r1/1:/M{in-ax}] new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 410702h14m1.480132072s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=2016-11-07 14:14:00.580657271 +0000 UTC]
W161107 14:14:00.581393 2301 storage/replica.go:1256  [s1,r1/1:/M{in-ax}] context canceled before command queue: Put ["one",/Min), Put ["two",/Min)
W161107 14:14:00.581932 2302 storage/replica.go:1765  [s1,r1/1:/M{in-ax}] ignoring cancelled context for command Put ["one",/Min), Put ["two",/Min)
panic: test timed out after 2m0s

goroutine 1693 [running]:
panic(0x16f0dc0, 0xc42011a030)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:918 +0x10b
created by time.goFunc
	/usr/local/go/src/time/sleep.go:154 +0x44

goroutine 1 [chan receive]:
testing.(*T).Run(0xc42018a300, 0x18ebfd3, 0x23, 0x1a9f348, 0xc42027fa01)
	/usr/local/go/src/testing/testing.go:647 +0x316
testing.RunTests.func1(0xc42018a300)
	/usr/local/go/src/testing/testing.go:793 +0x6d
testing.tRunner(0xc42018a300, 0xc4202b3bc0)
	/usr/local/go/src/testing/testing.go:610 +0x81
testing.RunTests(0x1aa1270, 0x238ef60, 0x154, 0x154, 0xc4202d1670)
	/usr/local/go/src/testing/testing.go:799 +0x2f5
testing.(*M).Run(0xc4202b3ee8, 0xc42027fd30)
	/usr/local/go/src/testing/testing.go:743 +0x85
github.com/cockroachdb/cockroach/pkg/storage_test.TestMain(0xc4202b3ee8)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/main_test.go:57 +0x1ff
main.main()
	github.com/cockroachdb/cockroach/pkg/storage/_test/_testmain.go:742 +0xc6

goroutine 17 [syscall, 2 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 2274 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe00000007)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc4201326a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 34 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).flushDaemon(0x26f8b00)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1001 +0x77
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:579 +0x95

goroutine 2254 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe00000003)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132620)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2256 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe00000005)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132660)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2287 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1133 +0x165
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132820)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2257 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe00000006)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132680)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2284 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).raftTickLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3255 +0x334
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420174cd0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2283 [semacquire]:
sync.runtime_Semacquire(0xc42018f8fc)
	/usr/local/go/src/runtime/sema.go:47 +0x30
sync.(*WaitGroup).Wait(0xc42018f8f0)
	/usr/local/go/src/sync/waitgroup.go:131 +0x97
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Wait(0xc42018f860)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:187 +0x34
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3229 +0x3a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420174cc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2249 [chan receive]:
github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).open.func1(0xc4206aec80)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:373 +0x44
created by github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).open
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:374 +0x702

goroutine 2282 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe0000000f)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc4201327a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2279 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe0000000c)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132740)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2285 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).startCoalescedHeartbeatsLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3286 +0x167
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420174cf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2250 [chan receive]:
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:171 +0x55
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc4201325a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2255 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe00000004)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132640)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2310 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).waitAndProcess(0xc42018fcc0, 0x7f2d37ee51c0, 0xc420012560, 0xecfb28328, 0x22a9d0c3, 0x26f7460, 0xc42000c240, 0xc4201acf30, 0xc42019fc00, 0xecfb28328)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:207 +0x254
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).scanLoop.func1.1(0xc42019fc00, 0xc42053f080)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:269 +0x8e
github.com/cockroachdb/cockroach/pkg/storage.(*storeReplicaVisitor).Visit(0xc42053f3b0, 0xc4205fc370)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:303 +0x3f5
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:271 +0x2ca
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420133c40)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2275 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe00000008)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc4201326c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2305 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:452 +0x357
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420133b60)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2311 [semacquire]:
sync.runtime_Semacquire(0xc42017457c)
	/usr/local/go/src/runtime/sema.go:47 +0x30
sync.(*Mutex).Lock(0xc420174578)
	/usr/local/go/src/sync/mutex.go:85 +0xd0
github.com/cockroachdb/cockroach/pkg/util/syncutil.(*TimedMutex).Lock(0xc42019fc78)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/timedmutex.go:84 +0x30
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).propose(0xc42019fc00, 0x7f2d37f650a0, 0xc42060ba70, 0x1484c97951929677, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1959 +0x1d3
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).tryAddWriteCmd(0xc42019fc00, 0x7f2d37f650a0, 0xc42060ba70, 0x1484c97951929677, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1746 +0x182
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).addWriteCmd(0xc42019fc00, 0x7f2d37f650a0, 0xc42060ba70, 0x1484c97951929677, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1651 +0x7a
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).Send(0xc42019fc00, 0x7f2d37f650a0, 0xc42060ba70, 0x1484c97951929677, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1113 +0x222
github.com/cockroachdb/cockroach/pkg/storage.(*pendingLeaseRequest).InitOrJoinRequest.func1(0x7f2d37f650a0, 0xc42060ba40)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_range_lease.go:132 +0x1df
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc4201acf30, 0x209af72, 0x1e, 0x96, 0x0, 0x0, 0xc42060b9e0, 0x7f2d37ee51c0, 0xc420012568)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:264 +0xdf
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:265 +0x234

goroutine 2302 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).tryAddWriteCmd(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6810, 0x1484c96b5950200f, 0x0, 0x0, 0x0, 0x0, 0x4045000000000000, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1755 +0x560
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).addWriteCmd(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6810, 0x1484c96b5950200f, 0x0, 0x0, 0x0, 0x0, 0x4045000000000000, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1651 +0x7a
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).Send(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6810, 0x1484c96b5950200f, 0x0, 0x0, 0x0, 0x0, 0x4045000000000000, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1113 +0x222
github.com/cockroachdb/cockroach/pkg/internal/client.Wrap.func1(0x7f2d37f65028, 0xc420123340, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4045000000000000, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/sender.go:77 +0xe9
github.com/cockroachdb/cockroach/pkg/internal/client.SenderFunc.Send(0xc420133ac0, 0x7f2d37f65028, 0xc420123340, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4045000000000000, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/sender.go:39 +0x75
github.com/cockroachdb/cockroach/pkg/storage.TestReplicaCommandQueueCancellation.func2.1(0x7f2d37ee51c0, 0xc420012560)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:2056 +0x512
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc4201acf30, 0x209b0a3, 0x17, 0x80a, 0x0, 0x0, 0xc420123380, 0x7f2d37ee51c0, 0xc420012560)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:264 +0xdf
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:265 +0x234

goroutine 2303 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:452 +0x357
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420133b20)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2277 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe0000000a)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132700)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2247 [chan receive]:
github.com/cockroachdb/cockroach/pkg/storage.TestReplicaCommandQueueCancellation(0xc42018a3c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:2108 +0x8eb
testing.tRunner(0xc42018a3c0, 0x1a9f348)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec

goroutine 2253 [chan send]:
github.com/cockroachdb/cockroach/pkg/storage.TestReplicaCommandQueueCancellation.func1(0x7f2d37f650a0, 0xc4208a6810, 0xc4205b4248, 0x8, 0x0, 0x1, 0x23b0980, 0xc42039a4e0, 0x1484c96b5950200f, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:2032 +0x7e
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).executeCmd(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6810, 0xc4205b4248, 0x8, 0x0, 0x7f2d37f63b00, 0xc42000ec40, 0xc420198700, 0x1484c96b5950200f, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:86 +0x38f4
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).executeBatch(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6810, 0xc4205b4248, 0x8, 0x7f2d37f63b00, 0xc42000ec40, 0xc420198700, 0x1484c96b5950200f, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3639 +0x426
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).executeWriteBatch(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6810, 0xc4205b4248, 0x8, 0x1484c96b5950200f, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3431 +0x1e0
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).applyRaftCommandInBatch(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6810, 0xc4205b4248, 0x8, 0x1484c96b5950200f, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3331 +0x127
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).evaluateProposal(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6810, 0x1, 0xc4205b4248, 0x8, 0x100000001, 0x1, 0x1484c96b5950200f, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1846 +0x148
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).processRaftCommand(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6810, 0xc4205b4248, 0x8, 0xe, 0x1, 0x100000001, 0x1, 0xc420198540, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3057 +0xa75
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).handleRaftReadyRaftMuLocked(0xc42019fc00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2312 +0x8a5
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).handleRaftReady(0xc42019fc00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2139 +0xcd
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processReady(0xc42019f180, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3174 +0x13e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:230 +0x28c
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132600)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2306 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:452 +0x357
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420133b80)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2307 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:452 +0x357
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420133ba0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2251 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xc400000010)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc4201325c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2308 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).waitAndProcess(0xc42018f9a0, 0x7f2d37ee51c0, 0xc420012560, 0xecfb28328, 0x22a94103, 0x26f7460, 0xc42000c240, 0xc4201acf30, 0xc42019fc00, 0xecfb28328)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:207 +0x254
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).scanLoop.func1.1(0xc42019fc00, 0xc42053f080)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:269 +0x8e
github.com/cockroachdb/cockroach/pkg/storage.(*storeReplicaVisitor).Visit(0xc42053f140, 0xc4205fc230)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:303 +0x3f5
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:271 +0x2ca
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420133bc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2248 [chan receive]:
github.com/cockroachdb/cockroach/pkg/rpc.NewContext.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:137 +0x76
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc4201e4b20)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2289 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).redirectOnOrAcquireLease(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6000, 0x620330)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:861 +0x502
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).getLeaseForGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3747 +0x6f
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0xc4201acf30, 0xc42040bcc0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:224 +0x100
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).getLeaseForGossip(0xc42019fc00, 0x7f2d37f650a0, 0xc4208a6000, 0xb, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3778 +0x112
github.com/cockroachdb/cockroach/pkg/storage.(*Store).maybeGossipSystemData(0xc42019f180, 0x7f2d37f650a0, 0xc4208a6000, 0x7d5911, 0x18)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1281 +0x160
github.com/cockroachdb/cockroach/pkg/storage.(*Store).(github.com/cockroachdb/cockroach/pkg/storage.maybeGossipSystemData)-fm(0x7f2d37f650a0, 0xc4208a6000, 0x0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1208 +0x3e
github.com/cockroachdb/cockroach/pkg/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1231 +0x232
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc42069a0f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2309 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:452 +0x357
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420133c20)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2280 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe0000000d)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132760)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2276 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe00000009)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc4201326e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2288 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).redirectOnOrAcquireLease(0xc42019fc00, 0x7f2d37f650a0, 0xc42069a210, 0x620330)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:861 +0x502
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).getLeaseForGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3747 +0x6f
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0xc4201acf30, 0xc4202adae8, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:224 +0x100
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).getLeaseForGossip(0xc42019fc00, 0x7f2d37f650a0, 0xc42069a210, 0xc4202adbb8, 0x10)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3778 +0x112
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).maybeGossipFirstRange(0xc42019fc00, 0x7f2d37f650a0, 0xc42069a210, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3817 +0x30c
github.com/cockroachdb/cockroach/pkg/storage.(*Store).maybeGossipFirstRange(0xc42019f180, 0x7f2d37f650a0, 0xc42069a210, 0x7d5911, 0x18)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1256 +0x1ac
github.com/cockroachdb/cockroach/pkg/storage.(*Store).(github.com/cockroachdb/cockroach/pkg/storage.maybeGossipFirstRange)-fm(0x7f2d37f650a0, 0xc42069a210, 0x0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1203 +0x3e
github.com/cockroachdb/cockroach/pkg/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1231 +0x232
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc42069a0c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2304 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:452 +0x357
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420133b40)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2252 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xc400000011)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc4201325e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2281 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe0000000e)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132780)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 2278 [semacquire]:
sync.runtime_notifyListWait(0xc42000db50, 0xe0000000b)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc42000db40)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42018f860, 0xc4201acf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:212 +0xb2
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4201acf30, 0xc420132720)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/replica_test.go
+++ b/pkg/storage/replica_test.go
@@ -2029,6 +2029,7 @@ func TestReplicaCommandQueueCancellation(t *testing.T) {
 	tsc.TestingKnobs.TestingCommandFilter =
 		func(filterArgs storagebase.FilterArgs) *roachpb.Error {
 			if filterArgs.Hdr.UserPriority == 42 {
+				log.Infof(context.Background(), "starting to block %s", filterArgs.Req)
 				blockingStart <- struct{}{}
 				<-blockingDone
 			}
@@ -2081,7 +2082,7 @@ func TestReplicaCommandQueueCancellation(t *testing.T) {
 
 	// Start a third command which will wait for the first.
 	ctx, cancel := context.WithCancel(context.Background())
-	cmd2Done := startBlockingCmd(ctx, key1, key2)
+	cmd3Done := startBlockingCmd(ctx, key1, key2)
 
 	// Wait until both commands are in the command queue.
 	util.SucceedsSoon(t, func() error {
@@ -2094,18 +2095,16 @@ func TestReplicaCommandQueueCancellation(t *testing.T) {
 		return nil
 	})
 
-	// Cancel the third command, then finish the first to unblock it.
+	// Cancel the third command; it will finish even though it is still
+	// blocked by the command queue.
 	cancel()
-	blockingDone <- struct{}{}
-	if pErr := <-cmd1Done; pErr != nil {
+	if pErr := <-cmd3Done; !testutils.IsPError(pErr, context.Canceled.Error()) {
 		t.Fatal(pErr)
 	}
 
-	// The third command should finish with a context cancellation
-	// error instead of executing. If it had started executing, it would
-	// be caught in the command filter and we'd time out reading from
-	// the channel.
-	if pErr := <-cmd2Done; !testutils.IsPError(pErr, context.Canceled.Error()) {
+	// Now unblock the first command to allow the test to clean up.
+	blockingDone <- struct{}{}
+	if pErr := <-cmd1Done; pErr != nil {
 		t.Fatal(pErr)
 	}
 }

